src/ngraph/runtime/cpu/builder/gather.cpp

```diff
@@ -29,116 +29,126 @@ namespace ngraph
-                        };
-                    }
-                    else
-                    {
-                        functor = [&,
-                                   params_shape,
-                                   indices_shape,
-                                   out_shape,
-                                   axis,
-                                   params_buffer_index,
-                                   indices_buffer_index,
-                                   out_buffer_index](CPURuntimeContext* ctx,
-                                                     CPUExecutionContext* ectx) {
-                            ngraph::runtime::reference::gather<double, int32_t>(
-                                static_cast<double*>(ctx->buffer_data[params_buffer_index]),
-                                static_cast<int32_t*>(ctx->buffer_data[indices_buffer_index]),
-                                static_cast<double*>(ctx->buffer_data[out_buffer_index]),
-                                params_shape,
-                                indices_shape,
-                                out_shape,
-                                axis);
-                        };
-                    }
+                    functor = prepare_functor<double>(node, args, out, external_function);
+                }
+                else if (element_type == element::i8)
+                {
+                    functor = prepare_functor<int8_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::i16)
+                {
+                    functor = prepare_functor<int16_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::i32)
+                {
+                    functor = prepare_functor<int32_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::i64)
+                {
+                    functor = prepare_functor<int64_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::u8)
+                {
+                    functor = prepare_functor<uint8_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::u16)
+                {
+                    functor = prepare_functor<uint16_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::u32)
+                {
+                    functor = prepare_functor<uint32_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::u64)
+                {
+                    functor = prepare_functor<uint64_t>(node, args, out, external_function);
+                }
+                else if (element_type == element::boolean)
+                {
+                    functor = prepare_functor<bool>(node, args, out, external_function);
```

Comment:
in ngraph element::boolean is not a c type bool, but is a `char` type.
Would be nice to have a unit test to check this out. Really does need a unit test for all types.

Reply:
OK. I will add these tests and change c type to char.

Reply:
It still looks like it is type bool

Reply:
Sorry, I missed that file in git add.